packages/backend-auth/src/userpool_access_policy_factory.ts

```diff
@@ -0,0 +1,122 @@
+import { IUserPool } from 'aws-cdk-lib/aws-cognito';
+import { Policy, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { Stack } from 'aws-cdk-lib';
+import { AmplifyFault, AmplifyUserError } from '@aws-amplify/platform-core';
+import { ActionIam, ActionMeta, AuthActions } from './types.js';
+
+/**
+ * Generates IAM policies scoped to a single userpool.
+ */
+export class UserPoolAccessPolicyFactory {
+  private readonly namePrefix = 'userpoolAccess';
+  private readonly stack: Stack;
+
+  private policyCount = 1;
+
+  /**
+   * Instantiate with the userpool to generate policies for
+   */
+  constructor(private readonly userpool: IUserPool) {
+    this.stack = Stack.of(userpool);
+  }
+
+  createPolicy = (actions: AuthActions) => {
+    const policyActions: Set<string> = new Set();
+
+    if (actions.length === 0) {
+      throw new AmplifyUserError('EmptyPolicyError', {
+        message: 'At least one action must be specified',
+      });
+    }
+
+    actions.forEach((authAction) => {
+      const mappedAction = iamActionMap[authAction];
+
+      if (typeof mappedAction === 'string') {
+        policyActions.add(mappedAction);
+      } else {
+        mappedAction.forEach((action) => policyActions.add(action));
+      }
+    });
+
+    if (policyActions.size === 0) {
+      throw new AmplifyFault('EmptyPolicyFault', {
+        message: 'Failed to construct valid policy to access UserPool',
+      });
+    }
+
+    const policy = new Policy(
+      this.stack,
+      `${this.namePrefix}${this.policyCount++}`,
+      {
+        statements: [
+          new PolicyStatement({
+            actions: [...policyActions],
+            resources: [this.userpool.userPoolArn],
+          }),
+        ],
+      }
+    );
+
+    return policy;
+  };
+}
+
+type IamActionMap = {
+  [action in ActionIam]: string;
+} & {
+  [amplifyActionSet in ActionMeta]: string[];
+};
+
+const iamActionMap: IamActionMap = {
+  manageUsers: [
+    'cognito-idp:AdminConfirmSignUp',
```

Comment:
I'd suggest taking this out and leaving it as a standalone action

Reply:
@edwardfoyle similar question but I feel this one should stay a part of "manageUsers" for manual confirmations

Reply:
My vote would still be to have this separate. Reading the docs, this doesn't seem like it would fall within normal "add/remove users" type scenarios, which is what I'm imagining for "manageUsers". It seems easy enough to add this as an additional permission if you want to do signup confirmations

Reply:
Chatted offline -- if we can confirm that this permission is needed when creating users, then I'm good with leaving this as part of the "manageUsers" grouping